dotcom-rendering/src/web/components/App.tsx

```diff
@@ -178,6 +182,10 @@ export const App = ({ CAPI, NAV, ophanRecord }: Props) => {
 		submitComponentEvent(componentEvent, ophanRecord);
 	};
 
+	const [asyncArticleCount, setAsyncArticleCount] = useState<
+		Promise<WeeklyArticleHistory | undefined>
```

Comment:
I like this. It's still a bit weird to pass a Promise into the components, but at least it contains a dependency (WeeklyArticleHistory).

We chatted about making the epic + banner slot selection into custom hooks, instead of components. App.tsx would render the results of these hooks (the winning epic/banner components). But this is a separate refactor.